src/Bicep.LangServer/Completions/BicepCompletionProvider.cs

```diff
@@ -872,9 +889,50 @@ private IEnumerable<CompletionItem> GetArrayItemCompletions(SemanticModel model,
 
         private IEnumerable<CompletionItem> GetFunctionParamCompletions(SemanticModel model, BicepCompletionContext context)
         {
-            if (!context.Kind.HasFlag(BicepCompletionContextKind.FunctionArgument) ||
-                context.FunctionArgument is null ||
-                model.GetSymbolInfo(context.FunctionArgument.Function) is not FunctionSymbol functionSymbol)
+            if (context.FunctionArgument is not { } functionArgument
+                || model.GetSymbolInfo(functionArgument.Function) is not FunctionSymbol functionSymbol)
+            {
+                return Enumerable.Empty<CompletionItem>();
+            }
+
+            //functionArgument flag indicates that we are about to type a function argument. But path completions should be also shown when on a argument value
```

Comment:
Did you explore using `TypeSymbolValidationFlags` instead of flags dedicated for functions? It would probably simplify a lot of this logic, because you wouldn't have to build your own overload-matching logic (`argType` on line 940 would give you this info).

Reply:
Hmm, didn't though about it... It seems that it would work, however the name - `ValidationFlags` doesn't seem correct and I didn't want to touch type system. However if you insist... :)